kilkari/src/main/java/org/motechproject/nms/kilkari/service/MotherCsvThreadProcessor.java

```diff
@@ -63,13 +64,30 @@ public ThreadProcessorObject call() throws Exception { //NOPMD SignatureDeclareT
             id = KilkariConstants.RCH_ID;
             contactNumber = KilkariConstants.MOBILE_NO;
             motherInstance = KilkariConstants.RCH_MOTHER;
+            mctsIDForRchMother = KilkariConstants.MCTS_ID;
         }
         int count = 0;
         Timer timer = new Timer("mom", "moms");
         for (Map<String, Object> record : recordList) {
             count++;
             LOGGER.debug("Started mother import for msisdn {} beneficiary_id {}", record.get(contactNumber), record.get(id));
 
+            //filter for  rch mother's Registration_No and mcts mother's ID_no
+            String newRchId=(String)record.get(id);
+            newRchId=newRchId.replaceAll("[\\n\\t\\r ]","");
+            record.replace(id,newRchId);
+            //for rch mother's MCTS_ID_No
+            if(!mctsImport){
+                try {
+                    String newMctsId=(String)record.get(mctsIDForRchMother);
+                    newMctsId=newMctsId.replaceAll("[\\n\\t\\r ]","");
+                    record.replace(mctsIDForRchMother,newMctsId);
+                }
+                catch (Exception e){
+                    //LOGGER.debug("no mcts id for the rch mother");
+                }
+            }
```

Comment:
Here as well, the common replace method can be used. Have this new method in a utility class and access everywhere.